compendium/DeclarativeServices/src/manager/states/CCActiveState.cpp

```diff
@@ -105,7 +105,16 @@ namespace cppmicroservices
                     // the latch counts down to 0, thereby allowing all Activate, Rebind and Modified
                     // activities to complete.
                     currentState->WaitForTransitionTask(); // wait for the previous transition to finish
-                    mgr.UnregisterService();
+                    try
+                    {
+                        mgr.UnregisterService();
+                    }
+                    catch (...)
+                    {
+                        mgr.GetLogger()->Log(cppmicroservices::logservice::SeverityLevel::LOG_WARNING,
+                                             "Failed to unregister the service.",
+                                             std::current_exception());
+                    }
```

Comment:
This makes sense to me - it means we'll definitely get to the next lines. Do we not want the same protection in `CCActiveState` in [this transition](https://github.com/CppMicroServices/CppMicroServices/blob/64860b15acd0dd17a234651e19a393d77de89f21/compendium/DeclarativeServices/src/manager/states/CCSatisfiedState.cpp#L47) ?

Reply:
Hmm good question. We didn't see this as a failure mode but that may just be a function of the test case. For that to happen we'd have to see this race while the component is satisfied but not active (concurrent shutdown and bundle stop of a satisfied but not active component). That seems like it could happen but is just less likely, probably why we haven't seen it as a failure mode. @jeffdiclemente what do you think about this